src/protocols/bdx/BdxEndpoint.h

```diff
@@ -0,0 +1,125 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ * @file BdxEndpoint.h
+ *
+ *  This file defines interfaces for connecting the BDX state machine (TransferSession) to the messaging layer.
+ */
+
+#include <core/CHIPError.h>
+#include <messaging/ExchangeContext.h>
+#include <messaging/ExchangeDelegate.h>
+#include <protocols/bdx/BdxTransferSession.h>
+#include <support/BitFlags.h>
+#include <system/SystemLayer.h>
+
+#pragma once
+
+namespace chip {
+namespace bdx {
+
+/**
+ * An abstract class with methods for handling BDX messages from an ExchangeContext and polling a TransferSession state machine.
+ *
+ * This class does not define any methods for beginning a transfer or initializing the underlying TransferSession object (see
+ * Initiator and Responder below).
+ * This class contains a repeating timer which regurlaly polls the TransferSession state machine.
+ * A CHIP node may have many Endpoints but only one Endpoint should be used for each BDX transfer.
+ */
+class Endpoint : public Messaging::ExchangeDelegate
```

Comment:
This naming is pretty confusing given the meaning of "Endpoint" in the interaction model.  Can we figure out a better name?

Reply:
Is it not enough that it is under the `bdx::` namespace?

This class can describe a BDX sender or receiver so I need a word that can be both. It is also an abstract class and is useless on its own. Its function is to integrate with the `ExchangeDelegate` API and regularly poll the `TransferSession` object. It also defines a method for implementing business logic.

- MessageHandler?
- TransferManager?
- TransferHandler?
- TransferDelegate?

I think I'd prefer TransferManager or TransferHandler

Reply:
Yes, either of those would work.  Or TransferParticipant?

Reply:
I like TransferParticipant. Too many "Managers" anyways

Reply:
Decided on `TransferFacilitator`